src/Orleans/Streams/Predicates/FilterPredicateWrapperData.cs

```diff
@@ -50,6 +50,9 @@ internal class FilterPredicateWrapperData : IStreamFilterPredicateWrapper
         [NonSerialized]
         private StreamFilterPredicate predicateFunc;
 
+        // constructor used by serializator
+        private FilterPredicateWrapperData() { }
```

Comment:
I hope not! That's one of the main advantages of Orleans's serialization - it just works, without any restrictions and special requirements. Plus compatible with .NET binary serialization, which of course does not require parameterless constructors.